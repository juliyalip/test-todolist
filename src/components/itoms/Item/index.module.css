.itemContainer {
  display: flex;
  border-radius: 4px;
  border-bottom: 0.5px solid var(--input-border-color);
  align-items: center;
  background-color: var(--primary-background-color);
  box-shadow: var(--medium-shadow);
  padding: 10px 12px;
  font-size: 0.9rem;
}
@media screen and (min-width: 768px) {
  .itemContainer {
    font-size: 1rem;
  }
}

.itemComplited {
  text-decoration: line-through;
  color: var(--complited-text-color);
}/*# sourceMappingURL=index.module.css.map */