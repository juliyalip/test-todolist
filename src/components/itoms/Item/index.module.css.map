{"version":3,"sources":["index.module.scss","index.module.css"],"names":[],"mappings":"AAAA;EACE,aAAA;EACA,kBAAA;EACA,oDAAA;EACA,mBAAA;EACA,iDAAA;EACA,gCAAA;EACA,kBAAA;EACA,iBAAA;ACCF;ADCE;EAVF;IAWI,eAAA;ECEF;AACF;;ADEA;EACE,6BAAA;EACA,kCAAA;ACCF","file":"index.module.css"}