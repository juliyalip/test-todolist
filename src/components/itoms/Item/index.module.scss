.itemContainer{
    display: flex;
    border-radius: 4px;
    border-bottom: 0.5px solid gray;
    align-items: center;
background-color: white;
   
}

.itemText{
    cursor: pointer;
    margin-left: 15px;
    display: flex;
    align-items: center;
    width: 100%;
   padding-right: 10px;
    justify-content: space-between;
    
   
}