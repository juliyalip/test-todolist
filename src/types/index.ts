export interface ITodo {
  text: string;
  id: string;
  done: boolean;
}

export interface INavigate {
  path: string;
  title: string;
}
