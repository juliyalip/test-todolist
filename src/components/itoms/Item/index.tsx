import Checkbox from '../Checkbox';
import Cross from '../Cross';
import { ITodo } from 'types';
import classnames from 'classnames';
import style from './index.module.scss';

interface ItemInterface {
  item: ITodo;
  onDelete: (id: string) => void;
  onComplited: (id: string) => void;
}

const Item = ({ item, onComplited, onDelete }: ItemInterface) => {
  return (
    <li className={style.itemContainer} >
      <Checkbox done={item.done} onComplited={onComplited} id={item.id} />

      <div className={style.itemText} data-testid="delete-todo" onClick={() => onDelete(item.id)}>
        <p
          className={classnames({
            [style.itemTextActive]: item.done,
          })}
        >
          {item.text}
        </p>

        {!item.done && <Cross />}
      </div>
    </li>
  );
};
export default Item;
