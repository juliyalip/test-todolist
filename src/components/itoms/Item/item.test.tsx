import { render } from '@testing-library/react';
import { ITodo } from 'types';
import Item from './index';

describe('test Item component', () => {
  const handleComplited = jest.fn();
  const handleDelete = jest.fn();

  it('prerender correctly', () => {
    const item: ITodo = { id: '001', text: 'I like Redux', done: false };
    render(
      <Item
        item={item}
        onComplited={handleComplited}
        onDelete={handleDelete}
      />,
    );
    const li = document.querySelector('li');
    const p = document.querySelector('p');
    expect(li).toBeInTheDocument();
    expect(p).toHaveTextContent('I like Redux');
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
});
