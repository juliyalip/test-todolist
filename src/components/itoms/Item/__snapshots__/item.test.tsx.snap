// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Item component prerender correctly 1`] = `
<p
  class=""
  data-testid="style"
>
  I like Redux
</p>
`;
